packages/browser-wallet/src/popup/store/utils.ts

```diff
@@ -80,3 +83,19 @@ export function atomWithChromeStorage<V>(key: ChromeStorageKey, fallback: V, wit
 
     return derived;
 }
+
+export function useListenForUpdates<V>(key: ChromeStorageKey, atomToUpdate: WritableAtom<V, V, void>) {
```

Comment:
There might be some infinite looping possible with this. I suggest that we improve our atoms with a way to make them listen to changes to the storage.

Reply:
I did this :tada: